File: 3 Веб Программирование/lab1/src/assets/php/AreaChecker.php
<?php

class AreaChecker{

    public static function inThisArea($x, $y, $r){

        //For triangle in top-left quadrant
        if($x <= 0 && $y >= 0){
            return ($x <= $r/2) && ($y <= $r) && (2 * $x + $y <= $r);
        }
        //For rectangle in bottom-left quadrant
        if($x < 0 && $y < 0){
            return ($x >= -$r) && ($y >= -$r);
        }
        //For circle in bottom-right quadrant
        if($x >= 0 && $y <= 0){
            return ($x * $x + $y * $y) <= ($r * $r);
        }
        
        //For top-right quadrant, always return false;
        return false;
    }

}

File: 3 Веб Программирование/labs/lab1/src/model/src/main/java/org/example/AreaChecker.java
package org.example;

public class AreaChecker {
    
    public boolean inArea(int x, Double y, Double r) {
        
        if (x <= 0 && y <= 0) {
            return (x >= -(r / 2)) && (y >= -r) && (- (2 * x) - y <= r);
        }
        // Rectangle in bottom-right quadrant
        if (x >= 0 && y <= 0) {
            return (x <= r / 2) && (y >= -r);
        }
        // Circle in top-left quadrant
        if (x <= 0 && y >= 0) {
            return (x * x + y * y) <= (r * r);
        }

        // For bottom-left quadrant, always return false
        return false;

    }

}